File: ДЗ 2/ch-4-1-step-11.swift
let target = Int(readLine()!)!

var remaining = target
var summands: [Int] = []

for number in 1... {
  let next = number + 1

  if (number + next) <= remaining {
    summands.append(number)
    remaining -= number
  } else {
    summands.append(remaining)
    remaining = 0
    break
  }
}

print(summands.count)
print(summands.map({ String($0) }).joined(separator: " "))

File: ДЗ 2/ch-4-1-step-11.py
target = int(input())

remaining = target
summands = []

for number in range(1, target+1):
    next_num = number + 1

    if (number + next_num) <= remaining:
        summands.append(number)
        remaining -= number
    else:
        summands.append(remaining)
        remaining = 0
        break

print(len(summands))
print(" ".join(str(i) for i in summands))
